### DISCUSSION TITLE
Maybe in 9 soft deleting could be called archiving

### DISCUSSION THREAD

**Discussion by dillingham** [laravel/framework#38780] [Ideas]
Kinda feels a little more refined:

- Post::withArchived()
- if($post->archived())
- delete -> archive()
- forceDelete -> delete()
- archived_at 
- public function archive() {}
- public function destroy() {}

**Comment by dennisprudlo** [comment]
I think there's a difference between deleting and archiving. In my case while soft-deleted models disappear entirely from the UI and are not likely to be restored, archived models can be found by the users in, well the archive.

**Reply by dillingham** [reply]
Yeah, I can see how they could coincide. In my case, an admin panel... the user deleting is the admin who would restore.. so it essentially is archiving. Viewing only deleted is viewing the archive
